### PR TITLE
6.1 account analytic project

### PR DESCRIPTION
**Pull request by NL66278**
Add possibility to create project from analytic account that was not a project before. Also provided link between analytic account and project,

### CODE REVIEW

File: account_analytic_project/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+##############################################################################
+{
+    'name': 'Project and analytic account integration',
+    'description': """\
+Project and analytic account integration.
+
+Gives the option to create projects directly from the analytic accounts they
+should be linked to.
+
+Withouth this module, if you first create an analytic account and later on you
```
Comment (hbrunn) [inline account_analytic_project/__openerp__.py]: -h

File: account_analytic_project/model/account_analytic_account.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        res = {}
+        for analytic_id in ids:
+            has_project = False
+            project_ids = project_model.search(
+                cr, uid, [('analytic_account_id', '=', analytic_id)],
+                context=context
+            )
+            if project_ids:
+                has_project = True
+            res[analytic_id] = has_project
```
Comment (hbrunn) [inline account_analytic_project/model/account_analytic_account.py]: why not bool(project_ids)?

**Comment by hbrunn** [comment]
some little nags, rest looks goot to me. One architectural question: Why don't you introduce the inverse to project.analytic_account_id here and use that in your checks? Then the ORM does all the searching for you, you don't need the function field and other code can simply loop through projects if necessary.

File: account_analytic_project/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Project and analytic account integration',
+    'description': """\
```
Comment (pedrobaeza) [inline account_analytic_project/__openerp__.py]: Remove \

File: account_analytic_project/i18n/account_analytic_project.pot [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: account_analytic_project
+#: code:addons/account_analytic_project/model/account_analytic_account.py:64
+#, python-format
+msgid "This method is only valid for a single analytic account"
+msgstr "This method is only valid for a single analytic account"
```
Comment (pedrobaeza) [inline account_analytic_project/i18n/account_analytic_project.pot]: pot file mustn't have strings on msgstr.

File: account_analytic_project/model/account_analytic_account.py (hunk 2 above)
Comment (pedrobaeza) [inline account_analytic_project/model/account_analytic_account.py]: And also a `break` to avoid consuming the rest of the loop

Comment (NL66278) [inline account_analytic_project/model/account_analytic_account.py]: But theoretically the method could be called for multiple rows. And if not, the loop will be done in one iteration anyway.

File: account_analytic_project/i18n/account_analytic_project.pot (hunk 4 above)
Comment (NL66278) [inline account_analytic_project/i18n/account_analytic_project.pot]: The pot file is the mostly unaltered result of exporting the English language from the module. That has all the msgstr's filled. Is this a problem? It is not a big problem to remove them from the few lines for this module, but other modules might have a lot of text, so If not needed, I would be glad to leave them the way they were exported.

Comment (pedrobaeza) [inline account_analytic_project/i18n/account_analytic_project.pot]: No, you can export an empty file if you select no language in the wizard. That's the correct pot file.

Comment (NL66278) [inline account_analytic_project/i18n/account_analytic_project.pot]: Thanks for the information. That was totally new for me.

**Comment by hbrunn** [comment]
[retracted thumbs up]

File: account_analytic_project/model/account_analytic_account.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        return {
+            'type': 'ir.actions.act_window',
+            'name': action_name,
+            'view_mode': view_mode,
+            'view_type': 'form',
+            'view_id': view_id,
+            'res_model': 'project.project',
+            'nodestroy': True,
+            'res_id': res_id,
+            'target':'current',
```
Comment (hbrunn) [inline account_analytic_project/model/account_analytic_account.py]: ./account_analytic_project/model/account_analytic_account.py:99:21: E231 missing whitespace after ':'

**Comment by hbrunn** [comment]
:+1: (flake8 and pylint errors are caused by other modules)

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**